Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp

```diff
@@ -347,61 +348,28 @@ static void webkitMediaCommonEncryptionDecryptProcessPendingProtectionEvents(Web
 
         GST_TRACE_OBJECT(self, "handling protection event %u for %s", GST_EVENT_SEQNUM(event.get()), eventKeySystemUUID);
 
-        if (priv->m_cdmInstance && g_strcmp0(eventKeySystemUUID, WebCore::GStreamerEMEUtilities::keySystemToUuid(priv->m_cdmInstance->keySystem()))) {
-            GST_TRACE_OBJECT(self, "protection event for a different key system");
+        GstMappedBuffer mappedBuffer(buffer, GST_MAP_READ);
+        if (!mappedBuffer) {
+            GST_WARNING_OBJECT(self, "cannot map protection data");
             continue;
         }
+        WebCore::InitData initData = WebCore::InitData(mappedBuffer.data(), mappedBuffer.size());
 
-        if (priv->m_currentEvent == GST_EVENT_SEQNUM(event.get())) {
-            GST_TRACE_OBJECT(self, "event %u already handled", priv->m_currentEvent);
+        if (priv->m_processedEventData.contains(initData)) {
+            GST_TRACE_OBJECT(self, "event %u already handled", GST_EVENT_SEQNUM(event.get()));
```

Comment:
This message is misleading, it should say something like "event %u init data already handled"